### DISCUSSION TITLE
Valid fields in a transmit completion

### DISCUSSION THREAD

**Discussion by traxys** [ofiwg/libfabric#9369] [Q&A] [answered]
From the man pages I'm not really sure what are the valid fields in a transmit completion (completion of a fi_write, fi_read, fi_send, ...),

For example should fields like:
- len
- buf
- data
- tag

Seem to be only related to receptions, so should they be filled in with the information passed by the user, or simply left empty?

**Comment by shefty** [comment, marked as answer]
The 4 fields mentioned are only valid for receive operations.  My recommendation is to set them to 0, but that's up to the provider. 
 Applications are not supposed to look at those fields for completed sends but leaving them uninitialized could result in debugging issues on the application side if it tries to interpret them.  Similarly, setting them to a valid value may result in an app assuming that all providers set those fields, which would lead to issues when swapping providers.
